src/chunk new.rs

```diff
@@ -0,0 +1,289 @@
+//! Chunk types and functions
+#![allow(dead_code)]
+// #![allow(non_upper_case_globals)]
+#![allow(non_camel_case_types)]
+
+macro_rules! chunk_types {
+    ($($chunk:ident[$type:expr]($size:expr)$(: $desc:literal)?)*) => {
+        #[derive(PartialEq, Eq, Clone, Copy)]
+        pub enum ChunkType {
+            $($chunk,)*
+            Unknown([u8; 4])
+        }
```

Comment:
I have to think about having an `enum` here. Png enourages handling all kinds of unknown chunks as long as the marker bits say that it is fine. What in particular is the added value of the enum over a struct with four chars?

Reply:
This enum is here just because when I thought of this I completely forgot about the decoding part of the library, that couldn't work with each chunk being a different type (at least without dyn dispatch). In the end I felt it wasn't all coming together as I thought it would have and dropped the idea, the only reason I left it here is because I thought that you might have seen it coming together in a different way